bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AbstractZkLedgerManager.java

```diff
@@ -61,7 +61,7 @@
 /**
  * Abstract ledger manager based on zookeeper, which provides common methods such as query zk nodes.
  */
-abstract class AbstractZkLedgerManager implements LedgerManager, Watcher {
+public abstract class AbstractZkLedgerManager implements LedgerManager, Watcher {
```

Comment:
Currently, from the `ScanAndCompareGarbageCollector` we need to get a ZK client instance to get the lock on the `ZkLedgerUnderreplicationManager`, so we're trying to reuse the same zk session from the `AbstractZkLedgerManager`. 

Maybe it would be better to just create a new session each time this scan runs? (given that the scan is meant to run very unfrequently, default 1day). With a new session we could avoid the downcasting at ScanAndCompareGarbageCollector.java:91